src/main/java/herrekt/Herrekt.java

```diff
@@ -0,0 +1,89 @@
+package herrekt;
+
+import herrekt.exceptions.InvalidInputException;
+import herrekt.exceptions.NoTaskException;
+import herrekt.taskmanager.*;
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+
+public class Herrekt {
+
+    private final Ui ui;
+    private final Storage storage;
+    private TaskList tasks;
+    private final Parser parser;
+
+    public Herrekt(String filePath) {
+        this.parser = new Parser();
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try {
+            this.tasks = new TaskList(this.storage.load());
+        } catch (FileNotFoundException e) {
+            this.ui.printLoadingError();
+            this.tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        this.ui.printWelcomeMessage();
+        Scanner sc = new Scanner(System.in);
+
+        String phrase = sc.nextLine();
+
+        while (!phrase.equals("bye")) {
+            try {
+                if (phrase.equals("list")) {
+                    ui.printTaskList(tasks);
+                } else if (phrase.contains("done")) {
+                    int taskNumber = parser.parseDoneInputToInt(phrase);
+                    tasks.markAsDone(taskNumber);
+                    ui.printMarkTaskAsDone(taskNumber, tasks);
+                } else if (phrase.contains("delete")) {
+                    int taskNumber = parser.parseDeleteInputToInt(phrase);
+                    ui.printTaskDeleted(taskNumber, tasks);
+                    tasks.delete(taskNumber);
+                } else {
+                    isInputValid(phrase);
```

Comment:
You check for valid phrases only after specific inputs like `"delete"` are not found, and rely on `catch` statements to handle the invalid phrase. This confuses [logical flow](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent). 

How about starting with checking that the phrase is valid and then tackling the specific cases? This will make the normal execution path more prominent.